### PR TITLE
Changed converting enum from and to raw values + added support for Ba…

### PR DESCRIPTION
**Pull request by Zazimou**
…ckedEnum in {default} property modifiers

### CODE REVIEW

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php [hunk 1]
```diff
@@ -30,8 +30,11 @@ public function convertToRawValue(mixed $value): mixed
 		if ($value === null) return null;
 		$type = array_key_first($this->propertyMetadata->types);
 		assert($value instanceof $type);
-		assert($value instanceof BackedEnum);
-		return $value->value;
+		if ($value instanceof BackedEnum) {
+			return $value->value;
+		}
```
Comment (hrach) [inline src/Entity/PropertyWrapper/BackedEnumWrapper.php:35]: What's the motivation here? Why should we allow not BackendEnum? If you don't need the conversion, is there any reason to use BackedEnumWrapper?

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php [hunk 2]
```diff
@@ -42,9 +45,16 @@ public function convertFromRawValue(mixed $value): ?BackedEnum
...
+		if (is_int($value) || is_string($value)) {
+			if (is_subclass_of($type, BackedEnum::class)) {
+				return $type::from($value);
+			}
+		}
+		if ($value instanceof BackedEnum && $value instanceof $type) {
+			return $value;
+		}
+
+		return null;
```
Comment (hrach) [inline src/Entity/PropertyWrapper/BackedEnumWrapper.php:58]: This seems dangerous to silently fallback to null.

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php [hunk 3]
```diff
@@ -42,9 +45,16 @@ public function convertFromRawValue(mixed $value): ?BackedEnum
...
-		assert(is_int($value) || is_string($value));
 		$type = array_key_first($this->propertyMetadata->types);
-		assert(is_subclass_of($type, BackedEnum::class));
-		return $type::from($value);
+		if (is_int($value) || is_string($value)) {
+			if (is_subclass_of($type, BackedEnum::class)) {
+				return $type::from($value);
+			}
+		}
+		if ($value instanceof BackedEnum && $value instanceof $type) {
```
Comment (hrach) [inline src/Entity/PropertyWrapper/BackedEnumWrapper.php:54]: Similar question, what's the motivation? What does it allow?

File: src/Entity/Reflection/ModifierParser.php [hunk 4]
```diff
@@ -190,29 +192,33 @@ private function processKeyword(string $value, ReflectionClass $reflectionClass)
...
-					if (substr($name, 0, $prefixLength) === $prefix) {
-						$enum[$constantValue] = $constantValue;
-						$count += 1;
-					}
-				}
-				if ($count === 0) {
-					throw new InvalidModifierDefinitionException("No constant matches $reflection->name::$const pattern.");
-				}
+			if ($reflection->isEnum() && is_subclass_of($className, BackedEnum::class)) {
+				return (new ReflectionEnum($className))->getCase($const)->getValue();
```
Comment (hrach) [inline src/Entity/Reflection/ModifierParser.php:196]: Please utilize the early return, the original code will stay (not indented).
Also please add a test.

**Comment by hrach** [comment]
cc @Zazimou

File: src/Entity/Reflection/ModifierParser.php (hunk 4 above)
Comment (Zazimou) [reply src/Entity/Reflection/ModifierParser.php:196]: done

**Comment by hrach** [comment]
Thanks, the bug is clear to me now, the fix is correct now, but I'll have to go over the test, ideally, I don't want to add more entities/tables to the test suit. Isn't possible to test it on existing infrastructure?

**Comment by datronjzima** [comment]
@hrach 

Tests reworked :-)

**Comment by datronjzima** [comment]
cc @hrach

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php (hunk 1 above)
Comment (hrach) [reply src/Entity/PropertyWrapper/BackedEnumWrapper.php]: irrelevant

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php (hunk 2 above)
Comment (hrach) [reply src/Entity/PropertyWrapper/BackedEnumWrapper.php]: irrelevant

File: src/Entity/PropertyWrapper/BackedEnumWrapper.php (hunk 3 above)
Comment (hrach) [reply src/Entity/PropertyWrapper/BackedEnumWrapper.php:57]: irrelevant

**Comment by hrach** [comment]
Could you please make the test pass?

**Comment by hrach** [comment]
Please "run `composer tests` locally (with Postgres) to generate the expected SQL queries files."

**Pull request merged**